src/editor.js

```diff
@@ -459,6 +544,72 @@ class Editor {
     }
     return str;
   }
+
+  /**
+   * Replaces CodeMirror hints with the AMP spec.
+   *
+   * Kinda stolen from @ampproject/docs/playground/src/editor.js
+   * (Editor#loadHints).
+   *
+   * Additionally sets uploaded file name hints for attrs like "src".
+   *
+   * @param {string} format
+   *    Should always be "amp4ads" for our use case, see top-level keys of
+   *    amphtml-hint.json for possible values.
+   * @private
+   */
+  async setHints_(format) {
+    const hints = await this.amphtmlHints_;
+    for (const key of Object.keys(codemirror.htmlSchema)) {
+      delete codemirror.htmlSchema[key];
```

Comment:
Do we need to delete if we are merging right after?

Reply:
Yeah, we get rid of the default HTML spec to replace it with AMPHTML's.

I'm not sure if there's a particular implementation detail for why the object is not simply replaced, since this was copied verbatim from `amp.dev`'s source. 

In any case, since this is exactly like the playground's code, maybe let's just leave it?

Reply:
Fine with leaving it.